lib/enum_state_machine/integrations/active_record.rb

```diff
@@ -455,12 +455,11 @@ def define_static_state_initializer
           # attributes passed into #initialize
           define_helper :class, <<-end_eval, __FILE__, __LINE__ + 1
             def column_defaults(*) #:nodoc:
-              super
-              @column_defaults = _default_attributes.deep_dup.to_hash
-              # No need to pass in an object, since the overrides will be forced
-              self.state_machines.initialize_states(nil, :static => :force, :dynamic => false, :to => @column_defaults)
-              @column_defaults.tap do |hash|
-                hash.freeze if Rails::VERSION::MAJOR >= 6
+              @column_defaults = super.tap do |defaults|
+                @column_defaults_unfrozen ||= defaults.frozen? ? defaults.deep_dup : defaults
```

Comment:
Apologies for this last minute change. I didn't like that I was using `_default_attributes` — an internal method which is most certainly subject to change in future Rails versions. I also didn't like that `deep_dup` was run every time this method was called, which was unnecessary and inefficient.